py3status/modules/imap.py

```diff
@@ -1,83 +1,94 @@
 # -*- coding: utf-8 -*-
-# FIXME new_mail_color param
 """
 Display number of unread messages from IMAP account.
 
 Configuration parameters:
-    cache_timeout: how often to run this check (default 60)
+    cache_timeout: refresh interval for this module (default 60)
     criterion: status of emails to check for (default 'UNSEEN')
-    format: format to display (default 'Mail: {unseen}')
-    hide_if_zero: don't show on bar if True (default False)
-    imap_server: IMAP server to connect to (default '<IMAP_SERVER>')
+    format: display format for this module (default 'Mail: {unseen}')
+    hide_if_zero: hide this module when no new mail (default False)
     mailbox: name of the mailbox to check (default 'INBOX')
-    new_mail_color: what color to output on new mail (default '')
-    password: login password (default '<PASSWORD>')
-    port: IMAP server port (default '993')
-    security: what authentication method is used: 'ssl' or 'starttls'
+    password: login password (default None)
+    port: number to use (default '993')
+    security: login authentication method: 'ssl' or 'starttls'
         (startssl needs python 3.2 or later) (default 'ssl')
-    user: login user (default '<USERNAME>')
+    server: server to connect (default None)
+    user: login user (default None)
 
 Format placeholders:
     {unseen} number of unread emails
 
+Color options:
+    color_new_mail: use color when new mail arrives (default None)
+
 @author obb
 """
-
 import imaplib
 from ssl import create_default_context
+STRING_UNAVAILABLE = 'Mail: N/A'
 
 
 class Py3status:
     """
     """
     # available configuration parameters
     cache_timeout = 60
+    color_new_mail = None
     criterion = 'UNSEEN'
     format = 'Mail: {unseen}'
     hide_if_zero = False
-    imap_server = '<IMAP_SERVER>'
     mailbox = 'INBOX'
-    new_mail_color = ''
-    password = '<PASSWORD>'
+    password = None
     port = '993'
     security = 'ssl'
-    user = '<USERNAME>'
+    server = None
+    user = None
+
+    class Meta:
+        deprecated = {
+            'rename': [
+                {
+                    'param': 'new_mail_color',
+                    'new': 'color_new_mail',
+                    'msg': 'obsolete parameter use `color_new_mail`',
+                },
+                {
+                    'param': 'imap_server',
+                    'new': 'server',
+                    'msg': 'obsolete parameter use `server`',
+                },
+            ],
+        }
 
     def post_config_hook(self):
         if self.security not in ["ssl", "starttls"]:
             raise ValueError("Unknown security protocol")
 
-    def check_mail(self):
+    def imap(self):
```

Comment:
again does this matter much we get all the info we need in the logs to know what was being called etc.